pages/search.js

```diff
@@ -0,0 +1,96 @@
+import Link from 'next/link'
+import { Prefetch } from '@layer0/react'
+import { useEffect, useState } from 'react'
+import LeftSidebar from '@/components/LeftSidebar'
+import RightSidebar from '@/components/RightSidebar'
+import { HeartIcon } from '@heroicons/react/outline'
+import { createNextDataURL } from '@layer0/next/client'
+
+const ProductPreview = ({ name, path, images, prices }) => {
+  const finalImage = `https://opt.moovweb.net/?quality=1&img=${images[0].url}`
+  const nonSlashPath = path.replace(/\//g, '')
+  const [productImage, setProductImage] = useState('/images/grey.png')
+  // In case JS is available, use the image loaded from the API response after the placeholders have been called
+  useEffect(() => {
+    let imagesList = window.imagesViewed
+    // If the visited object doesn't exist, create it
+    if (!imagesList) {
+      window.imagesViewed = {}
+      imagesList = window.imagesViewed
+    }
+    if (Object.keys(imagesList).length > 0 && imagesList[finalImage]) {
+      setProductImage(finalImage)
+    } else {
+      setTimeout(() => {
+        setProductImage(finalImage)
+        window.imagesViewed[finalImage] = true
```

Comment:
What's the purpose of this? Storing a global object shouldn't be necessary. There should be a better way to change a placeholder out. Can you just use a CSS background (just a grey image anyways)?